### PR TITLE
feat: Setup the Video Library

### PR DESCRIPTION
**Pull request by jalajcodes**
Its showing 10000+ lines added and 27k lines removed because I have removed the `package-lock.json` file and added `yarn.lock` file.

You don't need to review all files changed. Skip the css `styles` directory. And many components are repetitive so they won't take much time.

Created Responsive Navbar and sidebar.
Created required pages and styled them.

Things to review:

- `/pages` directory,  check any one file rest all are same for now.
- app.js
- `/components` directory
- `hooks/useLocationChange`
- **Skip** the `/styles` directory.

Live: https://max-video.herokuapp.com

### CODE REVIEW

File: frontend/src/App.js [hunk 1]
```diff
@@ -1,24 +1,43 @@
...
+import NotFound from "./components/NotFound";
+import Sidebar from "./components/Sidebar";
+import { useLocationChange } from "./hooks/useLocationChange";
+import Container from "./styles/Container";
+import History from "./pages/History";
+import Library from "./pages/Library";
+import YourVideos from "./pages/YourVideos";
+import LikedVideos from "./pages/LikedVideos";
+import Playlist from "./pages/Playlist";
+import Trending from "./pages/Trending";
```
Comment (vrinda-mahajan) [inline frontend/src/App.js]: Make an `index.js` file inside the component folder and one inside the pages folder and import and export all the components and pages inside that file and use that respective index.js file to import components from a single file.

File: frontend/src/components/Navbar.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+                height: 24,
+                marginTop: 8,
+              }}
+            />
+            <LogoText
+              style={{
+                width: 80,
+                height: 24,
+                marginLeft: -60,
+                marginTop: 5,
```
Comment (vrinda-mahajan) [inline frontend/src/components/Navbar.js]: Suggestion:- Try to avoid inline styling

File: frontend/src/components/Sidebar.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+} from "./Icons";
+import SidebarAuth from "./SidebarAuth";
+
+function Sidebar({ isSidebarOpen }) {
+  return (
+    <Wrapper open={isSidebarOpen}>
+      <NavLink
+        end
+        to="/"
+        className={({ isActive }) => (isActive ? "active" : "")}
```
Comment (vrinda-mahajan) [inline frontend/src/components/Sidebar.js]: Appreciation:- Good that you are using `Navlink` and `isActive` prop in the sidebar.

File: frontend/src/components/Sidebar.js [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+        to="/feed/playlist"
+        className={({ isActive }) => (isActive ? "active" : "")}
+      >
+        <div className="icon">
+          <SubIcon />
+          <span>Playlist</span>
+        </div>
+      </NavLink>
+
+      <div className="divider"></div>
```
Comment (vrinda-mahajan) [inline frontend/src/components/Sidebar.js]: Just a suggestion, you can also create an array and map through it for the sidebar component as the code is getting repeated

**Review by vrinda-mahajan — APPROVED** [review]
Added a few suggestions. Everything else looks good to me.👍

File: prisma/schema.prisma [hunk 5]
```diff
@@ -5,7 +5,7 @@ generator client {
 datasource db {
   provider          = "postgresql"
   url               = env("DATABASE_URL")
-  shadowDatabaseUrl = env("SHADOW_DATABASE_URL")
+  shadowDatabaseUrl = env("HEROKU_POSTGRESQL_CYAN_URL")
 }
 
 model User {
```
Comment (Dev-AkshatGupta) [inline prisma/schema.prisma:11]: Hey Jalaj
What's this file for? I don't know that's why asking

**Comment by Dev-AkshatGupta** [comment]
Everything else looks good to me keep on doing the good work.

File: prisma/schema.prisma (hunk 5 above)
Comment (jalajcodes) [reply prisma/schema.prisma:11]: So I am building the backend myself and I am using prisma, this is the schema file. Think of it as all the tables and columns present in the DB.

**Pull request merged**